From 4c68fa3d0e65e4f16f3fe43067427b42c5f07130 Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Wed, 13 Apr 2022 11:01:17 +0900
Subject: [PATCH 1/2] drivers/gicv3: Fix setting (E)PPI/SGI priorities

The second argument of gicr_write_ipriorityr specifies the interrupt ID.
This patch is modified to specify the interrupt ID from the register index.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
---
 drivers/arm/gic/v3/gicv3_helpers.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/arm/gic/v3/gicv3_helpers.c b/drivers/arm/gic/v3/gicv3_helpers.c
index 6bb66a02b..6de2fd2ca 100644
--- a/drivers/arm/gic/v3/gicv3_helpers.c
+++ b/drivers/arm/gic/v3/gicv3_helpers.c
@@ -267,7 +267,7 @@ void gicv3_ppi_sgi_config_defaults(uintptr_t gicr_base)
 	regs_num = ppi_regs_num << 3;
 	for (i = 0U; i < regs_num; ++i) {
 		/* Setup the default (E)PPI/SGI priorities doing 4 at a time */
-		gicr_write_ipriorityr(gicr_base, i, GICD_IPRIORITYR_DEF_VAL);
+		gicr_write_ipriorityr(gicr_base, i * 4, GICD_IPRIORITYR_DEF_VAL);
 	}
 
 	/* 16 interrupt IDs per GICR_ICFGR register */
-- 
2.36.1

